plasmapy/classes/plasma.py

```diff
@@ -34,46 +35,282 @@ class Plasma():
     pressure : `astropy.units.Quantity`
         (x, y, z) array of pressure at every point in the domain.
     magnetic_field : `astropy.units.Quantity`
-        (3, x, y, z) array of the magnetic field vector at every point in the domain.
+        (3, x, y, z) array of the magnetic field vector at every point in the
+        domain.
 
-        
     Parameters
     ----------
     domain_x : `astropy.units.Quantity`
-        1D array of x-coordinates for the plasma domain. Must have units convertable to length.
+        1D array of x-coordinates for the plasma domain. Must have units
+        convertable to length.
     domain_y : `astropy.units.Quantity`
-        1D array of y-coordinates for the plasma domain. Must have units convertable to length.
+        1D array of y-coordinates for the plasma domain. Must have units
+        convertable to length.
     domain_z : `astropy.units.Quantity`
-        1D array of z-coordinates for the plasma domain. Must have units convertable to length.
+        1D array of z-coordinates for the plasma domain. Must have units
+        convertable to length.
     """
     @u.quantity_input(domain_x=u.m, domain_y=u.m, domain_z=u.m)
-    def __init__(self, domain_x, domain_y, domain_z):
+    def __init__(self, domain_x, domain_y, domain_z, gamma=5/3):
         # Define domain sizes
         self.x = domain_x
         self.y = domain_y
         self.z = domain_z
 
         x, y, z = self.x.si.value, self.y.si.value, self.z.si.value
         self.grid = np.meshgrid(x, y, z, indexing='ij') * u.m
-        self.domain_shape = (len(self.x), len(self.y), len(self.z))
+        self.domain_shape = []
+        for length in (len(x), len(y), len(z)):
+            if length > 1:
+                self.domain_shape.append(length)
+        self.domain_shape = tuple(self.domain_shape)
+        print(self.domain_shape)
+        self.gamma = gamma
 
         # Initiate core plasma variables
-        self.density = np.zeros(self.domain_shape) * u.kg / u.m**3
-        self.momentum = np.zeros((3, *self.domain_shape)) * u.kg / (u.m**2 * u.s)
-        self.pressure = np.zeros(self.domain_shape) * u.Pa
-        self.magnetic_field = np.zeros((3, *self.domain_shape)) * u.T
+        self._density = np.zeros(self.domain_shape) * u.kg / u.m**3
+        self._momentum = np.zeros((3, *self.domain_shape)) * u.kg \
+            / (u.m**2 * u.s)
+        self._energy = np.zeros(self.domain_shape) * u.J / u.m**3
+        self._magnetic_field = np.zeros((3, *self.domain_shape)) * u.T
+
+        # Collect core variables into a list for usefulness
+        self.core_variables
+
+        # Connect a simulation object for simulating
+        self.simulation_physics = MHDSimulation(self)
```

Comment:
Doesn't that conflict with the idea of having multiple simulation engines (MHD, PIC etc) that can interact with the Plasma class?